sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/sin-validation-integration/_tests_/sin-validation-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  createFileFromStructuredRecords,
+  getStructuredRecords,
+  mockDownloadFiles,
+} from "@sims/test-utils/mocks";
+import { Job } from "bull";
+
+const SIN_VALIDATION_FILENAME = "PCSLP.PBC.BC0000.ISR";
+
+const padWithLeadingZeros = (num: number): string => {
+  // Pad the number with leading zeros to make it 9 digits long
+  return num.toString().padStart(9, "0");
+};
+
+describe(
+  describeProcessorRootTest(QueueNames.SINValidationResponseIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: SINValidationResponseIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+    let sinValidationResponseFolder: string;
+
+    beforeAll(async () => {
+      sinValidationResponseFolder = path.join(__dirname, "sin-receive-files");
+      process.env.ESDC_RESPONSE_FOLDER = sinValidationResponseFolder;
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      processor = app.get(SINValidationResponseIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+    });
+
+    it.only("should process SIN validation response file", async () => {
+      // Arrange
+      // Create a SIN record with REFERENCE_IDX = 600000001
+      const validSinStudent = await saveFakeStudent(db.dataSource, undefined, {
+        sinValidationInitialValue: {
+          sin: "100000001",
+          isValidSIN: true,
+        },
+      });
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [SIN_VALIDATION_FILENAME]);
+
+      mockDownloadFiles(
+        sftpClientMock,
+        [SIN_VALIDATION_FILENAME],
+        (fileContent: string) => {
+          const file = getStructuredRecords(fileContent);
+          const [record1] = file.records;
+
+          // Update the first record with validSinStudent's padded ID
+          const paddedId1 = padWithLeadingZeros(
+            validSinStudent.sinValidation.id,
+          );
+          file.records[0] =
+            record1.substring(0, 3) + paddedId1 + record1.substring(12);
+          return createFileFromStructuredRecords(file);
+        },
+      );
+
+      // Act
+      const processResult = await processor.processSINValidationResponse(job);
+      // Assert
+      const downloadedFile = path.join(
+        process.env.ESDC_RESPONSE_FOLDER,
+        SIN_VALIDATION_FILENAME,
+      );
+
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${downloadedFile}.`,
+            "File contains 2 SIN validations.",
+            "Processed SIN validation record from line 2: No SIN validation was updated because the record id is already present and this is not the most updated.",
```

Comment:
Please ensure the data setup will lead to the below message. It is going to a "non-happy path" probably because `dateReceived` is set.

This one the the happy, path.
```
SIN validation record updated.
```